pkg/daemon/ceph/client/pool.go

```diff
@@ -333,29 +333,49 @@ func setCommonPoolProperties(context *clusterd.Context, clusterInfo *ClusterInfo
 			}
 		}
 	} else {
-		if pool.Mirroring.Mode == "pool" {
-			// Remove storage cluster peers
-			mirrorInfo, err := GetPoolMirroringInfo(context, clusterInfo, pool.Name)
+		// Remove storage cluster peers
+		mirrorInfo, err := GetPoolMirroringInfo(context, clusterInfo, pool.Name)
```

Comment:
It looks like we're inside a pool reconcile. Why are we removing mirror peers during a pool reconcile? I thought this would be triggered by something else like a mirroring CR deletion.

Reply:
We set the mirror peer info in the `cephBlockPool` CR only.  That could be the reason why we are removing the peers during pool reconcile.  

The peers were set in the Mirror CR, but looks like that got [deprecated](https://github.com/rook/rook/blob/e41366bbcd9f38396db6a4caf44763d2c98f7f0b/pkg/operator/ceph/cluster/rbd/config.go#L84). And now they are set in the `cephblockpool` CR only.

Reply:
Let's keep that code path just for pool settings as much as possible. Looking [here](https://github.com/rook/rook/blob/master/pkg/operator/ceph/pool/controller.go#L342-L353) in the pool controller, how about if we reconcile the mirror peer cleanup there with that mirroring configuration?

Reply:
Done. Moved the change to the main pool controller reconcile when mirroring is disabled.